src/targets/gpu/device/reverse.cpp

```diff
@@ -0,0 +1,37 @@
+#include <migraphx/shape.hpp>
+#include <migraphx/argument.hpp>
+#include <migraphx/gpu/device/reverse.hpp>
+#include <migraphx/gpu/device/tensor.hpp>
+#include <migraphx/gpu/device/launch.hpp>
+#include <migraphx/gpu/device/types.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace gpu {
+namespace device {
+
+argument reverse(hipStream_t stream, argument result, argument arg1, int64_t axis)
+{
+    auto s                = arg1.get_shape();
+    auto axis_dim_size    = s.lens()[axis];
+    std::size_t nelements = s.elements();
+    visit_all(result, arg1)([&](auto output1, auto input1) {
+        hip_visit_views(output1, input1, s)([&](auto output, auto input, auto hs) {
+            // const auto* in_ptr = device_cast(input.data());
+            // auto* out_ptr = device_cast(output.data());
+            gs_launch(stream, nelements, 256)([=](auto i) __device__ {
```

Comment:
Is there a reason you need to set the local size to 256?

Reply:
Yea, defaulting to 1024 should better utilize the GPU. If you remove the parameter it will default to 1024.

Reply:
Okay makes sense, will replace it accordingly. Related, why does gather op use 256, any specific reason ?https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/0a9c18a582721085b211abb842c5b77c0fcce12b/src/targets/gpu/device/gather.cpp#L27) ?

Reply:
Changed it to 1024

Reply:
@cagery For whatever reason, 1024 for gather does not work correctly.